ultramodern/src/audio.cpp

```diff
@@ -65,3 +67,25 @@ uint32_t ultramodern::get_remaining_audio_bytes() {
     }
     return buffered_byte_count;
 }
+
+extern "C" u32 osAiGetStatus() {
+    uint32_t cur_frame_count = 0;
+    if (audio_callbacks.get_frames_remaining != nullptr) {
+        cur_frame_count = audio_callbacks.get_frames_remaining();
+    }
+
+    uint32_t samples_per_vi = (sample_rate / 60);
+    if (cur_frame_count >  static_cast<uint32_t>(buffer_offset_frames * samples_per_vi)) {
+        cur_frame_count -= static_cast<uint32_t>(buffer_offset_frames * samples_per_vi);
+    }
+    else {
+        cur_frame_count = 0;
+    }
+
+    // Check if the most recently queued samples are playing, if not then consider the DMA as incomplete.
+    if (cur_frame_count > recent_frames_queued.load()) {
```

Comment:
Shouldn't you be updating `recent_frames_queued` somewhere? Otherwise this will always eval to zero

Reply:
You're right, and upon further inspection adding the missing piece to this implementation breaks the recomp this was used for. It doesn't make sense to merge this if it's incomplete, so I'll close it for now.